__tests__/tests/components/CommentInput.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import {Platform} from 'react-native';
+
+import {CommentInput} from 'components';
+
+describe('<CommentInput />', () => {
+
+  let defaultProps = {
+    users: [],
+    userHasPushPermission: true,
+    issueLocked: false,
+    locale: '',
+    onSubmit: () => {},
+  };
+
+  it("should render 'TextInput' and TouchableOpacity if user can post", () => {
+
+    const wrapper = shallow(
+      <CommentInput {...defaultProps}/>
+    );
+
+    expect(wrapper.find('TextInput').length).toEqual(1);
+    expect(wrapper.find('TouchableOpacity').length).toEqual(1);
+    expect(wrapper.find('Text').length).toEqual(0);
+  });
+
+  it("should not render 'TextInput' and TouchableOpacity if user can post", () => {
+
+    const wrapper = shallow(
+      <CommentInput
+        {...defaultProps}
+        userHasPushPermission={false}
+        issueLocked={true}/>
```

Comment:
Shall we separate as 2 cases? For `userHasPushPermission` and `issueLocked`.

Reply:
I think no. Because `userHasPushPermission ` and `issueLocked` are linked with the operation and.
![image](https://user-images.githubusercontent.com/5528931/32133075-8c3cc0a4-bbc8-11e7-91a3-5a5995dc138a.png)

Reply:
@dyesseyumba Yes, there are 2 kinds of results. But we have 4 kinds of inputs, `userHasPushPermission` x2 and `issueLocked` x2. And `issueLocked` also affects the locked issue text and icon. Can you cover them?

Reply:
Okay, I'm on it.